src/services/user.service.ts

```diff
@@ -1,49 +1,62 @@
+import { Op } from "sequelize";
+
 import { IUser } from "../models";
-import { UserRepository } from "../repositories";
+import { User } from "../models";
 
 export class UserService {
-  constructor(private readonly userRepository: UserRepository) {}
-
-  getUsers(): IUser[] {
-    return this.userRepository.getUsers();
+  async getUsers(): Promise<User[]> {
+    return await User.findAll({ where: { isDeleted: false } });
   }
 
-  getUserByID(userId: string): IUser {
-    return this.userRepository.getUserByID(userId);
-  }
+  async getUserByID(userId: string): Promise<User> {
+    const user = await User.findOne({
+      where: { id: userId, isDeleted: false },
+    });
 
-  addUser(user: IUser): IUser {
-    return this.userRepository.addUser(user);
+    if (!user) {
+      throw `User with id=${userId} is not found`;
+    }
+
+    return user;
   }
 
-  updateUser(userId: string, user: IUser): IUser {
-    return this.userRepository.updateUser(userId, user);
+  async addUser(user: IUser): Promise<User> {
+    await this.checkUserLogin(user);
+    const newUser = await User.create(user);
+    return newUser;
   }
 
-  deleteUserById(id: string): void {
-    this.userRepository.deleteUserById(id);
+  async updateUser(userId: string, user: IUser): Promise<User> {
+    await this.checkUserLogin(user, userId);
+    await User.update(user, { where: { id: userId } });
+    return await this.getUserByID(userId);
   }
 
-  getAutoSuggestUsers(loginSubstring: string, limit: number): IUser[] {
-    const users = this.userRepository.getUsers();
+  async deleteUserById(userId: string): Promise<void> {
+    await User.update({ isDeleted: true }, { where: { id: userId } });
+  }
 
-    return users
-      .filter((user) => user.login.includes(loginSubstring))
-      .sort(this.sortUsersBy("login"))
-      .slice(0, limit);
+  async getAutoSuggestUsers(
+    loginSubstring: string,
+    limit: number
+  ): Promise<User[]> {
+    return await User.findAll({
+      where: { isDeleted: false, login: { [Op.like]: `%${loginSubstring}%` } },
+      order: ["login"],
+      limit,
+    });
   }
 
-  private sortUsersBy(
-    property: string | number
-  ): (user1: IUser, user2: IUser) => number {
-    return (user1: IUser, user2: IUser) => {
-      if (user1[property] < user2[property]) {
-        return -1;
-      } else if (user1[property] > user2[property]) {
-        return 1;
-      } else {
-        return 0;
-      }
-    };
+  private async checkUserLogin(user: IUser, userId?: string): Promise<void> {
+    const userWithSameLogin = await User.findOne({
```

Comment:
I would add the getUserByLogin method. For example, It could be useful for authorization